src/ipred_prepare.rs

```diff
@@ -139,6 +139,21 @@ pub unsafe fn rav1d_prepare_intra_edges<BD: BitDepth>(
     bd: BD,
 ) -> IntraPredMode {
     let bitdepth = bd.bitdepth();
+    let stride = BD::pxstride(stride as usize) as isize;
+    let first_pos = if have_left { -1 } else { 0 } + if have_top { -stride } else { 0 };
+    let last_pos = if have_left {
+        (4 * h - 1) as isize * stride
+    } else if have_top {
+        4 * w as isize
+    } else {
+        0
+    };
+    let dst = if last_pos > first_pos {
+        slice::from_raw_parts(dst.offset(first_pos), (last_pos - first_pos) as usize)
+    } else {
+        &[]
+    };
+    let dst_origin = (-first_pos) as usize;
```

Comment:
What if `first_pos` is positive, like if `stride` is negative?  I know I have to re-design the `BD::pxstride` API, but assuming that's fixed.

Reply:
Also, similar to `topleft_out_offset`, I think `dst_offset` makes more sense here.  Also, I think it may be clearer to calculate

```rust
let dst_offset = if have_left { 1 } else { 0 } + if have_top { stride } else { 0 };
```

instead of a negative `first_pos`.  Then you'd do `dst.sub(dst_offset)` instead of `dst.offset(first_pos)` and `dst_offset + last_pos` instead of `last_pos - first_pos`.